test/test_mqtt.py

```diff
@@ -213,5 +213,111 @@ def on_sub_message(topic, payload):
         connection.disconnect().result(TIMEOUT)
 
 
+class MqttConnectionSingletonTest(NativeResourceTest):
+    TEST_TOPIC = '/test/me/senpai'
+    TEST_MSG = 'NOTICE ME!'.encode('utf8')
+
+    def _create_connection(self, auth_type=AuthType.CERT_AND_KEY):
+        config = Config(auth_type)
```

Comment:
[style] try to re-use more code.

instead of copy/pasting 200 lines of code, you could modify the `def _create_connection()` helper function with an additional arg. And just 1 test new actual test case like `test_connect_disconnect_with_static_singletons` is fine

```
def _create_connection(self, auth_type=AuthType.CERT_AND_KEY, use_static_singletons=False):
    bootstrap = None
    if not use_static_singletons:
        elg = EventLoopGroup()
        resolver = DefaultHostResolver(elg)
        bootstrap = ClientBootstrap(elg, resolver)
    ...        
```